torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp

```diff
@@ -340,20 +340,27 @@ void PrepareCopyForONNX(Node* node) {
     // tensor indices input is empty.
     // Remove aten::copy_, and replace it with index_put.
     // 1. create an empty listConstruct node as indices input for index_put.
-    // 2. create index_put node.
+    // 2. cast expanded_value, so it has the same type as node->input[0].
+    // 3. create index_put node.
 
     // Tracing aten::copy_ broadcasts the rhs values.
-    // 3. Apply broadcasting for scripting.
+    // 4. Apply broadcasting for scripting.
     WithInsertPoint guard(node);
     auto graph = node->owningGraph();
     auto dummy_list =
         graph->insertNode(graph->createList(OptionalType::ofTensor(), {}))
             ->output();
 
+    auto* cast = graph->create(::c10::onnx::Cast, 1);
+    cast->addInput(node->input(1));
+    auto aten_type = node->input(0)->type()->cast<TensorType>()->scalarType();
```

Comment:
need to add check for 1. if `cast<TensorType>()` returns nullptr, and 2. if `scalarType()` returns c10::nullopt.